client/client.go

```diff
@@ -1403,16 +1403,32 @@ func (c *Client) blockForRemoteAlloc(alloc *structs.Allocation) {
 		c.migratingAllocsLock.Unlock()
 	}()
 
-	// Block until the previous allocation migrates to terminal state
-	prevAlloc, err := c.waitForAllocTerminal(alloc.PreviousAllocation)
-	if err != nil {
-		c.logger.Printf("[ERR] client: error waiting for allocation %q: %v", alloc.PreviousAllocation, err)
+	// If the allocation is not sticky then we won't wait for the previous
+	// allocation to be terminal
+	tg := alloc.Job.LookupTaskGroup(alloc.TaskGroup)
```

Comment:
Need to mark the allocation as failed. This is what the alloc runner does:
`r.setStatus(structs.AllocClientStatusFailed, fmt.Sprintf("missing task group '%s'", alloc.TaskGroup))`